First DSL2 version

KevinMenden opened the pull request:
## PR checklist

- [ X] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated
- [ ] If necessary, also make a PR on the [nf-core/cageseq branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/cageseq)

This is a first working DSL2 version of the cageseq pipeline.
The **CTSS_QC** is not working yet due to a missing multi-tool container, so it has to be run using `--skip_ctss_qc`
We might also opt for two separate processes here, instead of relying on a multi-tool container (bedtools, rseqc).

The pipeline now works with a samplesheet-based input (`--input samplesheet.csv`).

There are still some minor changes necessary, e.g. to the modules.config and to go over some of the processes, but I want to put it here so it's easier to work on it.

apeltzer commented:
> The CTSS_QC is not working yet due to a missing multi-tool container, so it has to be run using --skip_ctss_qc

In case you were wondering how to do that => https://github.com/BioContainers/multi-package-containers :-) Its really simpler than I thought in the beginning 😓

KevinMenden commented:
Yes, created a PR yesterday - it's almost too simply to be true :D am waiting for someone to accept the PR.

KevinMenden commented:
Alright I fixed all warnings, fixed issues with version and multiqc collections and added modules to the modules.config file - should be a solid first version.
The PR for the multi-tool container has been merged but I still can't find the container - might take some time.
The PR for adding the samplesheet to the test-datasets has been reviewed, but not merged, so until this is done the tests can't work :)
At least the linting is passing ;)

mashehu commented on modules/local/process/bowtie_align.nf:
```diff
@@ -0,0 +1,46 @@
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    // Note: 2.7X indices incompatible with AWS iGenomes.
```
copy-paste error from the STAR process 🙂

KevinMenden replied:
Oops :) a lot of copy-pasting was involved, I admit it

mashehu commented on modules/local/process/bowtie_align.nf:
```diff
@@ -0,0 +1,46 @@
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    // Note: 2.7X indices incompatible with AWS iGenomes.
+    conda (params.enable_conda ? "bioconda::bowtie=1.2.3" : null)
```
I just saw that bowtie 1.3.0 is out. Since we are already at it, should we update to it?

KevinMenden replied:
Yeah, good point. Let's update it.
I will do some local testing as well on monday to get the tests running, and hopefully make some final commits to this PR then.

KevinMenden commented:
Updated the bowtie version to 1.3.0 and fixed some typos in the test configuration files.

Locally, running with the test profile works nicely. However all tests are failing here.
Apparently the reason is a compilation problem in `process_ctss.nf` - but I don't get it, because that file looks fine, and I should get the same error locally.

Any ideas?

mashehu commented on nextflow.config:
```diff
@@ -154,7 +154,7 @@ manifest {
   mainScript = 'main.nf'
   nextflowVersion = '>=20.07.1'
-  version = '1.0.0'
+  version = '2.0.0dev'
```
for the version bump you should run `nf-core bump-version . 2.0.0dev`

mashehu commented on Dockerfile:
```diff
@@ -7,9 +7,9 @@ COPY environment.yml /
+RUN conda env export --name nf-core-cageseq-2.0.0dev > nf-core-cageseq-2.0.0dev.yml
 
 # Instruct R processes to use these empty files instead of clashing with a local version
 RUN touch .Rprofile
```
Could we not use biocontainers for all our things and with that get rid of this Dockerfile and environment.yml?

KevinMenden replied:
Yeah had them only in there because I wasn't sure how we'll go ahead with that ....
Simply deleted them now.

mashehu commented on cageseq.nf:
```diff
@@ -0,0 +1,262 @@
+
+// Get rRNA databases
+// Default is set to bundled DB list in `assets/rrna-db-defaults.txt`
+ribo_database = file(params.ribo_database_manifest)
```
This whole part doesn't have to be run if `remove_ribo_rna` is set. Put it in an if-clause.

mashehu commented on cageseq.nf:
```diff
@@ -0,0 +1,262 @@
+            ch_gtf
+        )
+        ch_bam = ALIGN_BOWTIE.out.bam
+        // ch_software_versions = ch_software_versions.mix(ALIGN_BOWTIE.out.bowtie_version.first().ifEmpty(null))
```
```suggestion
```

KevinMenden replied:
there was a module-related issue with the version naming, fixed this now.

KevinMenden commented:
Made the bowtie alignment process more similar to the STAR process, they both now use the same samtools sub-workflow which is nice.
This also adds more QC to the output when using either of the two aligners.
